File: iosApp/iosApp/Firebase/FirebaseStorage.swift
import FirebaseStorage
import Foundation

class FirebaseStorage {
    private static let storage = Storage.storage()
    private static let reference = storage.reference()
    
    func getData(_ child: String, _ maxSize: Int64 = 1024 * 1024, callback: @escaping (Data?) -> Void) {
        FirebaseStorage.reference.child(child).getData(maxSize: maxSize) { result, error in
            if let error = error {
                callback(nil)
            }
            else {
                if let result = result {
                    callback(result)
                }
                else {
                    callback(nil)
                }
            }
        }
    }
}

File: androidApp/src/main/java/com/cistus/hunter/android/firebase/FirebaseStorage.kt
package com.cistus.hunter.android.firebase

import com.google.firebase.ktx.Firebase
import com.google.firebase.storage.ktx.storage

class FirebaseStorage {
    companion object {
        private val storage = Firebase.storage
        private val reference = storage.reference
    }

    fun getData(child: String, maxSize: Long = 1024 * 1024, callback: (ByteArray?) -> Unit) {
        reference.child(child).getBytes(maxSize)
            .addOnSuccessListener {
                if (it.isEmpty())
                    callback(null)
                else
                    callback(it)
            }
            .addOnFailureListener { callback(null) }
    }
}